### PR TITLE
(EWL-897): New login modal UI - Mobile / Desktop / Tablet 🖥📱

### PR DESCRIPTION
**Pull request by JulienKode**
# [(EWL-897)](https://eurosportdigital.atlassian.net/browse/EWL-897): New login modal UI - Mobile / Desktop / Tablet 🖥📱

Please check the type of changes you're about to make

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- Browser compatibility checked:
  - [x] Safari
  - [x] Chrome
  - [x] Firefox
- [x] MR title respects the format: `(EWL-XXX): description`
- [x] Move Jira ticket into PR

### Login

|         Mockup         |      New component     |  
| :--------------------: | :--------------------: | 
| Mockup | Login | 

## Example

|         Login         |   
| :--------------------: | 
| example |

## Description

you can test the Login component here: https://deploy-preview-263--es-web-toolkit.netlify.com

- [x] create Login component
- [x] handle submit by forwarding user credentials
- [x] handle update of credentials in the input
- [x] create other UI component 
- [x] create `UnderlineLink` component
- [x] add snapshot testing for `Login` component
- [x] add snapshot testing for `UnderlineLink` component
- [x] test expected behaviour of the `Login` component
- [x] use `Input` from #253 
- [x] use `ErrorBanner` from #255   
- [x] use `Button` with type form from #257  
- [x] use `ArrowLink` from #37 
- [x] use `UnderlineLink` from #264 

## Test Plan

- setup a desktop stories
- create a mobile stories 
- use them both with responsive on chrome, firefox and safari

### CODE REVIEW

File: src/modules/Login/index.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  emailPlaceholder,
+  signInText,
+  showPasswordText,
+  title,
+  errorMessage,
+}) => {
+  const [password, setPassword] = useState('');
+  const [email, setEmail] = useState('');
+
+  const onEmailChange = useCallback(({ target }) => setEmail(target.value), [setEmail]);
```
Comment (saad-kh) [inline src/modules/Login/index.js]: Hoooooks!
Nice but Maybe we can start using the following eslint rules:
```
"react-hooks/rules-of-hooks": "error",
"react-hooks/exhaustive-deps": "warn",
```

File: src/modules/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  const onPasswordChange = useCallback(({ target }) => setPassword(target.value), [setPassword]);
+  const onSignInSubmit = useCallback(() => onSubmit({ email, password }), [onSubmit]);
+
+  return (
+    <MainContainer>
```
Comment (saad-kh) [inline src/modules/Login/index.js]: I'm from the school that it's better to still use a form dom tag for semantics and better browser behavior :)

File: src/modules/Login/index.js (hunk 1 above)
Comment (saad-kh) [reply src/modules/Login/index.js]: nice use of the useCallback by the way

File: src/modules/Login/index.js (hunk 2 above)
Comment (JulienKode) [reply src/modules/Login/index.js]: Behind the button implementation it's `a` is it ok with that ?

File: src/modules/Login/index.js (hunk 1 above)
Comment (JulienKode) [reply src/modules/Login/index.js]: You mean implementing them ? @saad-kh

Comment (saad-kh) [reply src/modules/Login/index.js]: No just install `eslint-plugin-react-hooks` and add the two rules I specified in `.eslintrc`

File: src/modules/Login/index.js (hunk 2 above)
Comment (saad-kh) [reply src/modules/Login/index.js]: Yes, just capture the onSubmit of the form, make sure there's no post action on it, and use this event to login.
Forms give us better semantics, better SEO, better browser behaviors, better form validation etc.

File: src/modules/Login/index.js (hunk 1 above)
Comment (JulienKode) [reply src/modules/Login/index.js]: Thank you @saad-kh

File: src/modules/Login/index.js (hunk 2 above)
Comment (JulienKode) [reply src/modules/Login/index.js]: Yes but onSubmit is called on a button with `input` of type `submit`

we do not have one here as `Button` are an implementation of `a`

- https://github.com/EurosportDigital/web-toolkit/blob/master/src/elements/Button/index.js#L6
- https://github.com/EurosportDigital/web-toolkit/blob/master/src/elements/ArrowLink/index.js#L14

File: src/modules/Login/index.js (hunk 1 above)
Comment (JulienKode) [reply src/modules/Login/index.js]: b943276c876dc97074f2ff440566a38315fa0b23 should solve this

File: src/modules/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
+import React, { useState, useCallback } from 'react';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+import Input from '../../components/Input/input.component';
```
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/modules/Login/index.js]: b45f6d068106eba76f22c351f61cc1b9e658ef85 should solve this

File: src/elements/UnderlineLink/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import Link from '../Link';
```
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/elements/UnderlineLink/index.js]: You're right, absolute path is better for that case 

9f7e6348998da464257a7332c3c359d1e554ac70 should solve this

**Comment by JulienKode** [comment]
@saad-kh Thanks for your review, I've made an update

File: src/modules/Login/index.js (hunk 2 above)
Comment (JulienKode) [reply src/modules/Login/index.js]: 7ad0b492140a80cace81aa11285fdcfd47c08fe1 should solve this issue

**Pull request merged**